### ISSUE TITLE
[Feature]: Page report breakdown of TVL by protocol

### ISSUE THREAD

**Issue by storming0x** [yearn/yearn-watch#8]
Currently the risk dashboard has impact grouping based on smart contract groups per strategy.

We also need a report page where we can see allocations per protocol aggregated.

Would be great to have a page that can answer questions like, how much exposure overall yearn has in XYZ protocol?

**Comment by Space-zee** [comment]
Hello, I`d like to take this task. Is it still relevant? If so, can you give me more details?
@storming0x   @magic990619

**Comment by storming0x** [comment]
> Hello, I`d like to take this task. Is it still relevant? If so, can you give me more details? @storming0x @magic990619

Yes i think this is still valuable @Majorfi can also review details

**Comment by Space-zee** [comment]
@Majorfi , please ,can you give me more details about this task

**Comment by Majorfi** [comment]
Hey @Space-zee !
You can check [this](https://d3971bp2359cnv.cloudfront.net/api) Risk Framework API to get some info.
You can get the TVL per strategy per group the same way we are doing it in the risk page, [here](https://github.com/yearn/yearn-watch/blob/main/pages/risk.tsx#L98-L114)

The idea is to get a good visualisation of the breakdown per TVL. Both a pie chart + a list just like in [this](https://yearn.watch/track) page bellow could be good (Strategy Name, TVL, Rank).

**Comment by bogdanuch41** [comment]
@Majorfi I'm working together with Space-zee. I wanted to ask: what exactly is expected to be seen? From what I understand - each strategy is connected to 2 protocols. So I should add the same TVL to each protocol? Because I think that layout will be similar to Strategy layout. And what is meant by Rank param that you have written?

**Comment by storming0x** [comment]
>each strategy is connected to 2 protocols
@bogdanuch41 not necessarily the case, it could be the just 1 protocol.

Ideally we have a dashboard with bar graph that answers the following questions:

- How many funds does yearn have allocated t in protocol xyz (e.g AAVE) across all chains and strategies? 
- What about just Ethereum, arbitrum, etc? (filter)
- How many strategies use protocol X  (aggregation of all strategies across all vaults)? list the strategies, drill down into specific strategies and the existing navigation strategies like strategy page and vault page

**Comment by bogdanuch41** [comment]
@Majorfi @storming0x  ![image](https://user-images.githubusercontent.com/92035951/184878410-086cced0-adbf-401f-981e-e37105a8b6ad.png)
Hello, so I've made it like this. I wanted to ask you if it's okay and if you have any other params that you want to get, because I couldn't think of any? Also maybe there is some image that I can use for this page link(or I can find something in Web).

**Comment by storming0x** [comment]
Awesome!, i would add a  bar chart of allocations in % (similar to what we have in debt ratio in strategy page) on total yearn TVL, and the 100% should cover every network.

Would that make sense?

**Comment by bogdanuch41** [comment]
@storming0x. Sorry, I'm not sure which bar chart you're referring to. Do you mean something like this?
![image](https://user-images.githubusercontent.com/92035951/185347284-ea1f4dd5-0ca9-4ab8-b976-8db13526cc18.png)

**Comment by bogdanuch41** [comment]
Hello @storming0x . Can you give your answer about bar chart please? Or maybe I can make pool request without it?

**Comment by storming0x** [comment]
> Hello @storming0x . Can you give your answer about bar chart please? Or maybe I can make pool request without it?

hi, the one in vault page breakdown which has all strategy allocations
https://yearn.watch/vault/0x0fBbf9848D969776a5Eb842EdAfAf29ef4467698
<img width="546" alt="Screen Shot 2022-08-22 at 16 09 06" src="https://user-images.githubusercontent.com/6074987/186027386-b5811951-858d-4f03-8384-ccdae700dc09.png">

we prefer bar chart instead of pie if possible

**Comment by bogdanuch41** [comment]
@storming0x hello again. Is it okay like this? 
![image](https://user-images.githubusercontent.com/92035951/186715793-ca4928e7-344c-4281-bcc7-76a5ac4a6084.png)
I also added a Allocated strategies amount(amount of strategies with tvl>0), since in cases of small numbers 0% showed tvl allocation can have some tvl(like strategy with tvl of 10 in 10000000 tvl protocol)

**Comment by storming0x** [comment]
> @storming0x hello again. Is it okay like this? ![image](https://user-images.githubusercontent.com/92035951/186715793-ca4928e7-344c-4281-bcc7-76a5ac4a6084.png) I also added a Allocated strategies amount(amount of strategies with tvl>0), since in cases of small numbers 0% showed tvl allocation can have some tvl(like strategy with tvl of 10 in 10000000 tvl protocol)

hey, yeah this looks like a good dashboard to drill down, i think having the  bar chart  also be global would help

curveFinance % overall TVL
convex and 88, etc 

and from that bar chart you can drill down to this page or link i guess

**Comment by bogdanuch41** [comment]
@storming0x not sure, what did you mean. There is division of data by chain and also the sum of all chains in All data like this. 
![image](https://user-images.githubusercontent.com/92035951/186800164-02a45100-3ab4-44b7-8547-0f6b7020b5ea.png)
Did you mean that I should add a column like this 
![image](https://user-images.githubusercontent.com/92035951/186800409-428e24a3-88a2-47eb-aa41-38581ff71b7c.png)

**Comment by storming0x** [comment]
hope the quick drawing helps
![ywatch bar chart](https://user-images.githubusercontent.com/6074987/186935502-48a23fd3-ca57-4e86-a2ed-ca5be67b6b05.png)

@bogdanuch41 so essentially a visual graph works best to see overall exposure taken from the data you have on current screen and can have two views, one overview graph and one detail like the one above for data, makes sense?

**Comment by bogdanuch41** [comment]
![image](https://user-images.githubusercontent.com/92035951/187169316-a7b6e6b9-057b-4cd1-97d3-08885f38dcec.png)
@storming0x  if I undertand you correctly, then this is already implemented? This percent is showing how much of total tvl for this chain(or for all) is located in this protocol.

**Comment by storming0x** [comment]
> n this is already implemented? This percent is showing how much of total tvl for this chain(or for all) is located in this protocol.

Awesome, i didnt see the % before or understood that was the same breakdown, yeah this should work

**Comment by bogdanuch41** [comment]
@storming0x, I have made the pool request now, so if you are free - you can look into it.

**Comment by bogdanuch41** [comment]
@storming0x , @Majorfi , I've seen that my PR wasn't merged and that it have merge issues, so I resolved them. Is it going to be merged now?

**Comment by bogdanuch41** [comment]
Sorry, @Majorfi, I'm not sure if PR comment answers ping user, so I'll also write this here: I refactored code as you required and wanted to know is formating in pages/allocations.ts good now?
